docs/proposals/deploy-csi-to-k3s.md

```diff
@@ -0,0 +1,54 @@
+# Proposal: Deploy CSI on K3s cluster 
+
+Last updated: 8.04.22
+
+
+## Abstract
+
+Deploy CSI on K3s cluster, test it work and add steps for manual patching scheduler 
+
+## Background
+
+Currently csi is compatible with Vanilla Kubernetes, RKE2, OpenShift and Kind, but only for test purpose. We want to deploy csi to new platform.  
+
+## Proposal
+
+Need to create next config file in directory `/var/lib/rancher/k3s/agent/pod-manifests`:
+```
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+extenders:
+  - urlPrefix: "http://127.0.0.1:8889"
+    filterVerb: filter
+    prioritizeVerb: prioritize
+    weight: 1
+    enableHTTPS: false
+    nodeCacheCapable: false
+    ignorable: true
+    httpTimeout: 15s
+leaderElection:
+  leaderElect: true
+clientConnection:
+  kubeconfig: /var/lib/rancher/k3s/server/cred/scheduler.kubeconfig
+```
+For passing it to k3s server component use `--kube-scheduler-arg=config`
+
+
+## Compatibility
+
+There is no problem with compatibility
+
+## Implementation
+
+Tasks:
+1.	Create new config for scheduler 
```

Comment:
In my opinion, we can automate config creation and k3s.service file modifying in scheduler-patcher. But now it has option just to work with yaml files contained kubernetes manifests - https://github.com/dell/csi-baremetal/blob/f9a3d599f9dcaf715bbc54f952d45e3b37d4ea62/pkg/scheduler/patcher/main.py#L205
Could we parse `/etc/systemd/system/k3s.service` in python code or is it not an option?

Reply:
I found how they create this [file ](https://github.com/k3s-io/k3s/blob/master/install.sh#L766) and suppose we can parse it.

Reply:
Decided that scheduler patcher is not needed. User will have to change configuration manually.